### DISCUSSION TITLE
Are foreign tables with postgres_fdw supported?

### DISCUSSION THREAD

**Discussion by wvengen** [mathesar-foundation/mathesar#3414] [User Q&A] [answered]
Hi, I've just started exploring this project, and setup a test instance. The database I'm connecting to is a new database with a number of foreign tables using [postgres_fdw](https://www.postgresql.org/docs/current/postgres-fdw.html).

After setting up the foreign tables, I can query them using `psql`.
After selecting _Sync External Changes_ in Mathesar, the new schema appears but no tables are seen.

Is Mathesar supposed to work with foreign tables, or is this an unsupported feature?

In a related question, I tried working with views, but they don't show up in Mathesar either. Not sure about their status (https://github.com/mathesar-foundation/mathesar/issues/442 didn't help me much).

**Comment by kgodey** [comment, marked as answer]
@wvengen We have not yet tested Mathesar with foreign tables, so I would consider them unsupported for now. If you'd like us to add support in a future version, please open an issue.

We do not yet support reflecting views, but we have already received a couple of requests for this and we plan to build support for this. We do not have a GitHub issue for it yet. We're working on building out a roadmap that makes our future plans more visible to the community.

I apologize for #442 being confusing. Our original plan was to build our visual query builder on top of database views, but that implementation proved to be untenable for a few different reasons. The main reason was that we could not accurately reflect queries built using the UI since Postgres doesn't store the query in its original form in all cases. This meant the user would not be able to edit the query that a given view was built on. So we ended up coming up with the Exploration / Data Explorer concept for our query builder and pushing support for database views to later. #442 reflected our initial plan, which we later changed.

**Comment by wvengen** [comment]
Thank you for responding and explaining how things went, much appreciated! Very clear. We'll see if it is suitable enough for us now, and if so, would consider contributing.

**Reply by kgodey** [reply]
You're welcome! If you do end up using Mathesar, we'd love to know more about your use case. And we always welcome new contributors!
